htdocs/compta/accounting-files.php

```diff
@@ -240,9 +251,10 @@
 			$sql .= " FROM ".MAIN_DB_PREFIX."payment_various as t";
 			$sql .= " WHERE datep between ".$wheretail;
 			$sql .= " AND t.entity IN (".$db->sanitize($entity == 1 ? '0,1' : $entity).')';
+			if(!empty($projectid)) $sql .= " AND fk_projet = ".$db->sanitize($projectid);
 		}
 		// Loan payments
-		if (GETPOST('selectloanspayment') && !empty($listofchoices['selectloanspayment']['perms'])) {
+		if (GETPOST('selectloanspayment') && !empty($listofchoices['selectloanspayment']['perms']) and empty($projectid)) {
```

Comment:
For code standardization, use the "&&" for the and in PHP. We use the "AND" for the and of SQL.

Reply:
Oops, the AND is a reflex from a pythonista :D
Fixed!